atacworks/io/bw2h5.py

```diff
@@ -0,0 +1,148 @@
+        cleanpeakbw: BigWig file containing clean peaks data to be used as
+        labels for training.
+        batch_size: Number of lines to read at a time, since all of the data
+        does not fit in memory.
+        nolabel: No peak labels to encode into h5. Used for generating
+        data to
+        run inference on.
+        nonzero: Only save intervals that have non-zero values.
+        intervals_file: File containing the intervals corresponding to the
+        training,
+        validation or inference.
+        out_dir: Directory to save the output files to.
+        prefix: Prefix to attach to the name, to make the files unique.
+        pad: Padding values.
+
+    Returns:
+        Path to output files.
+
+    """
+    # Read intervals
+    _logger.info('Reading intervals')
+    intervals = read_intervals(intervals_file)
+    _logger.info('Read {} intervals'.format(len(intervals)))
+
+    # Create output directory
+    if not os.path.exists(out_dir):
+        os.makedirs(out_dir)
+    # Optionally, select intervals with nonzero coverage
+    if nonzero:
+        _logger.info('Selecting intervals with nonzero coverage')
+        nonzero_intervals = check_bigwig_intervals_nonzero(
+            intervals, noisybw)
+        _logger.info("Retaining {} of {} nonzero noisy intervals".format(
+            sum(nonzero_intervals), len(intervals)))
+        intervals = intervals[nonzero_intervals]
+
+    _logger.debug('Collecting %d intervals' % len(intervals))
+
+    # Calculate number of batches
+    batches_per_epoch = int(np.ceil(len(intervals) / batch_size))
+    _logger.info('Writing data in ' + str(batches_per_epoch) + ' batches.')
+
+    # Split intervals into batches
+    batch_starts = np.array(range(0, len(intervals), batch_size))
+    batch_ends = batch_starts + batch_size
+    batch_ends[-1] = len(intervals)
+
+    # Get output hdf5 filename
+    output_file_path = os.path.join(out_dir, prefix + '.h5')
+
+    # Write batches to hdf5 file
+    _logger.info('Extracting data for each batch and writing to h5 file')
+    for i in range(batches_per_epoch):
+
+        # Print current batch
+        if i % 10 == 0:
+            _logger.info("batch " + str(i) + " of " + str(batches_per_epoch))
+
+        # Create dictionary to store data
+        batch_data = {}
+
+        # Subset intervals
+        batch_intervals = intervals.iloc[batch_starts[i]:batch_ends[i], :]
+
+        # Read noisy data
+        batch_data['input'] = extract_bigwig_intervals(
+            batch_intervals, noisybw, pad=pad
+        )
+
+        # Add other input layers
+        if layersbw is not None:
+            # Read additional layers
+            layers = gather_key_files_from_cmdline(layersbw, extension='.bw')
+            for key in layers.keys():
+                batch_data[key] = extract_bigwig_intervals(
+                    batch_intervals, layers[key], pad=pad
+                )
+
+        # Add labels
+        if not nolabel:
```

Comment:
Do we need the nolabel option? Can we assume that if no label files are given we should encode only the noisy data?

Reply:
removed